src/client/Client.cc

```diff
@@ -10326,26 +10326,22 @@ void Client::ll_register_callbacks(struct client_callback_args *args)
 
 int Client::test_dentry_handling(bool can_invalidate)
 {
+  ceph_assert((can_invalidate && dentry_invalidate_cb) || remount_cb);
+
   int r = 0;
 
   can_invalidate_dentries = can_invalidate;
 
   if (can_invalidate_dentries) {
-    ceph_assert(dentry_invalidate_cb);
     ldout(cct, 1) << "using dentry_invalidate_cb" << dendl;
     r = 0;
   } else if (remount_cb) {
     ldout(cct, 1) << "using remount_cb" << dendl;
     r = _do_remount(false);
   }
   if (r) {
-    bool should_abort = cct->_conf.get_val<bool>("client_die_on_failed_dentry_invalidate");
-    if (should_abort) {
-      lderr(cct) << "no method to invalidate kernel dentry cache; quitting!" << dendl;
-      ceph_abort();
-    } else {
-      lderr(cct) << "no method to invalidate kernel dentry cache; expect issues!" << dendl;
-    }
+    lderr(cct) << "no method to invalidate kernel dentry cache; expect issues!"
+               << dendl;
```

Comment:
This code is now only reachable if `_do_remount` fails. The warning would indicate that it didn't even try to do remount.

I think we can just remove the message and ignore errors from `_do_remount`. `_do_remount` already aborts if `client_die_on_failed_dentry_invalidate` is true.

Reply:
makes sense -- will update